### PR TITLE
Add preview mode

### LINKED ISSUE #123: Pulumi preview
Since Defang uses pulumi under the hood to manage stacks, etc. it would be really useful to be able to just run `defang pulumi <pulumi commands>`

### PR DESCRIPTION
**Pull request by lionello**
Fixes #123 by adding a new `defang cd preview` command. Technically, this command should live under `defang compose` because it needs the Compose file, but it's BYOC only and shows raw Pulumi output, so for pros only.

If we're happy with this, we can "mirror" `preview` to top-level, like `defang preview` cf. `defang up` etc.

### CODE REVIEW

File: src/cmd/cli/command/commands.go [hunk 1]
```diff
@@ -148,15 +149,16 @@ func SetupCommands(version string) {
 	_ = RootCmd.MarkPersistentFlagFilename("file", "yml", "yaml")
 
 	// Bootstrap command
```
Comment (jordanstephens) [inline src/cmd/cli/command/commands.go:151]:
```suggestion
	// cd command
```

**Review by jordanstephens — APPROVED** [review]

File: src/pkg/cli/composeUp.go [hunk 2]
```diff
@@ -49,7 +39,7 @@ func ComposeUp(ctx context.Context, c client.Client, force bool, mode defangv1.D
 		return nil, project, &ComposeError{fmt.Errorf("no services found")}
 	}
 
-	if DoDryRun {
+	if force == compose.BuildContextIgnore {
```
Comment (edwardrf) [inline src/pkg/cli/composeUp.go:42]: Should we rename the variable `force`? it is a bit confusing

**Review by edwardrf — APPROVED** [review]
LGTM

File: src/pkg/cli/composeUp.go (hunk 2 above)
Comment (lionello) [reply src/pkg/cli/composeUp.go:42]: good call

**Pull request merged**